src/engagement_db_to_analysis/traced_data_filters.py

```diff
@@ -65,45 +65,41 @@ def rqa_time_range_filter(user, messages_traced_data, pipeline_config):
     return filtered
 
 
-def filter_test_participants(user, participants_traced_data_map, test_participant_uuids):
+def filter_test_participants_messages(user, messages_traced_data, test_participant_uuids):
     """
     Filters a dict of participants who are not in pipeline_config.test_contacts e.g AVF/Aggregator staff
 
-    :param participants_traced_data_map: A dict of TracedData individuals objects to filter.
-    :type participants_traced_data_map: dict of TracedData
+    :param messages_traced_data: List of message objects to filter.
+    :type messages_traced_data: list of TracedData
     :param test_participant_uuids: a list containing test participant uids.
     :type test_participant_uuids: list of str
     :return: Filtered dict.
     :rtype: dict of participant_uid -> TracedData
     """
     log.debug("Filtering out test participants data...")
-    filtered = {}
+    filtered = []
 
-    for uid, participant_trace_data in participants_traced_data_map.items():
-        if uid in test_participant_uuids:
+    for td in messages_traced_data:
+        if td["participant_uuid"] in test_participant_uuids:
             continue
 
-        participant_trace_data.append_data(participant_trace_data, Metadata(user, Metadata.get_call_location(),
+        td.append_data(td, Metadata(user, Metadata.get_call_location(),
```

Comment:
Can you add a comment here explaining why you're appending the td to itself, as it's not obvious on first reading

Reply:
updated thanks.